tasks/actorstate/miner/sector_deals.go

```diff
@@ -23,21 +24,29 @@ func (SectorDealsExtractor) Extract(ctx context.Context, a actorstate.ActorInfo,
 		span.SetAttributes(a.Attributes()...)
 	}
 
-	ec, err := NewMinerStateExtractionContext(ctx, a, node)
+	ec, err := extraction.LoadMinerStates(ctx, a, node)
 	if err != nil {
 		return nil, fmt.Errorf("creating miner state extraction context: %w", err)
 	}
 
-	var sectors []*miner.SectorOnChainInfo
-	if !ec.HasPreviousState() {
+	return ExtractSectorDealsModel(ctx, ec)
+}
+
+func ExtractSectorDealsModel(ctx context.Context, ec extraction.State) (minermodel.MinerSectorDealList, error) {
```

Comment:
Move the model extraction logic to a testable method. `extraction.State` is an interface we can mock.